lib/Dialect/FIRRTL/Export/FIREmitter.cpp

```diff
@@ -401,8 +410,8 @@ void Emitter::emitModulePorts(ArrayRef<PortInfo> ports,
     const auto &port = ports[i];
     ps << (port.direction == Direction::In ? "input " : "output ");
     if (!arguments.empty())
-      addValueName(arguments[i], port.name);
-    ps << PPExtString(port.name.getValue()) << " : ";
+      addValueName(arguments[i], legalize(port.name));
+    ps << PPExtString(legalize(port.name)) << " : ";
```

Comment:
(maybe not worth only calling `legalize(port.name)` once, as done elsewhere/later? Works for me!)